Resolved issue54, counter validate attribute

mhs5218 opened the pull request:
counter/_counter.scss  
- Added counter-error class, used when displaying validator messages

counter.module.js
- Added 4th counter to demonstrate custom validators on counters

counter.template.html
- Added blt-validate attribute to counter input (needed for data-counter to work)
- Added error message when custom validator returns false

mhs5218 committed:
counter/_counter.scss
   *Added counter-error class, used when displaying validator messages

counter.module.js
   *Added 4th counter to demonstrate custom validators on counters

counter.template.html
   *Added blt-validate attribute to counter input (needed for data-counter to work)
   *Added error message when custom validator returns false

jrh418 commented:
There's a bug with the display / validation here. I've seen 3 things:
- If you use the down button and reach 0, it will disallow inputting a new value or using the arrow buttons
- If you input a value directly, it will change to 0 and disallow inputting a new value or using the arrow buttons
- It will sometimes shift the number down (adds a padding) when the value is invalid & one of the two above situations are true

mhs5218 committed:
Removed bug where numbers sometimes shift down when the value is invalid.
Removed bug that disallowed inputting a new value or using the arrow buttons when the counter reached 0.
Removed bug where inputting a value directly also resulted in the counter's value getting assigned to 0.
